### DISCUSSION TITLE
Si training RMS errors question

### DISCUSSION THREAD

**Discussion by Mskater** [deepmodeling/deepmd-kit#807] [Q&A]
Hi,
Do these look right for Si training RMS errors?  Purple is energy, green is force.
Thanks

![Si_energy_force_RMSerrors](https://user-images.githubusercontent.com/71668612/123683348-194a9680-d81a-11eb-8313-604b2311eebf.png)

Here is the training input file:
{
    "_comment": " model parameters",
    "model": {
	"type_map":	["Si"],
	"descriptor" :{
	    "type":		"se_a",
	    "sel":		[432],
	    "rcut_smth":	5.80,
	    "rcut":		6.00,
	    "neuron":		[25, 50, 100],
	    "resnet_dt":	false,
	    "axis_neuron":	16,
	    "seed":		1,
	    "_comment":		" that's all"
	},
	"fitting_net" : {
	    "neuron":		[240, 240, 240],
	    "resnet_dt":	true,
	    "seed":		1,
	    "_comment":		" that's all"
	},
	"_comment":	" that's all"
    },

    "learning_rate" :{
	"type":		"exp",
	"decay_steps":	5000,
	"start_lr":	0.001,	
	"stop_lr":	3.51e-8,
	"_comment":	"that's all"
    },

    "loss" :{
	"start_pref_e":	0.02,
	"limit_pref_e":	1,
	"start_pref_f":	1000,
	"limit_pref_f":	1,
	"start_pref_v":	0,
	"limit_pref_v":	0,
	"_comment":	" that's all"
    },

    "_comment": " training controls", 
    "training" : {
	"systems":	["../Si/"],
	"set_prefix":	"set",    
	"stop_batch":	1000000,
	"batch_size":	"auto",

	"seed":		1,

	"_comment": " display and restart",
	"_comment": " frequencies counted in batch",
	"disp_file":	"lcurve.out",
	"disp_freq":	100,
	"numb_test":	10,
	"save_freq":	1000,
	"save_ckpt":	"model.ckpt",
	"load_ckpt":	"model.ckpt",
	"disp_training":true,
	"time_training":true,
	"profiling":	false,
	"profiling_file":"timeline.json",
	"_comment":	"that's all"
    },

    "_comment":		"that's all"
}

**Comment by tuoping** [comment]
```json
"sel": [432],
"rcut_smth": 5.80,
"rcut": 6.00,
```
These settings are not good. rcut_smth is too large, to close to rcut. In addition, sel is too large. Please refer to the example setting provided with package. https://github.com/deepmodeling/deepmd-kit/blob/devel/examples/water/se_e2_a/input.json

**Reply by Mskater** [reply]
Thank you @tuoping .
I tried Si training again, and got this RMS errors graph in training.  I am running v1.3.3.
And the input file is attached too.   How to tell if this is right from looking at the graph?
Thanks, as I am just getting started.

![SiRMSerrors](https://user-images.githubusercontent.com/71668612/124492637-bc536100-dd82-11eb-9042-42fc1aeefd2e.png)

{
"_comment": " model parameters",
"model": {
"type_map": ["Si"],
"descriptor" :{
"type": "se_a",
"sel": [46],
"rcut_smth": 0.5,
"rcut": 6.00,
"neuron": [25, 50, 100],
"resnet_dt": false,
"axis_neuron": 16,
"seed": 1,
"_comment": " that's all"
},
"fitting_net" : {
"neuron": [240, 240, 240],
"resnet_dt": true,
"seed": 1,
"_comment": " that's all"
},
"_comment": " that's all"
},

"learning_rate" :{
"type":		"exp",
"decay_steps":	5000,
"start_lr":	0.001,	
"stop_lr":	3.51e-8,
"_comment":	"that's all"
},

"loss" :{
"start_pref_e":	0.02,
"limit_pref_e":	1,
"start_pref_f":	1000,
"limit_pref_f":	1,
"start_pref_v":	0,
"limit_pref_v":	0,
"_comment":	" that's all"
},

"_comment": " training controls", 
"training" : {
"systems":	["../Si/"],
"set_prefix":	"set",    
"stop_batch":	1000000,
"batch_size":	"auto",

"seed":		1,

"_comment": " display and restart",
"_comment": " frequencies counted in batch",
"disp_file":	"lcurve.out",
"disp_freq":	100,
"numb_test":	10,
"save_freq":	1000,
"save_ckpt":	"model.ckpt",
"load_ckpt":	"model.ckpt",
"disp_training":true,
"time_training":true,
"profiling":	false,
"profiling_file":"timeline.json",
"_comment":	"that's all"
},

"_comment":		"that's all"
}

**Comment by tuoping** [comment]
Sorry for the late reply, some thing went wrong with my email and I have been on it. 
"sel": [46] <---this setting is clearly not large enough. There must be WARNINGS printed on screen or in the returned file, please take a look. A recommended setting is "sel: [300]".
In addition, how large is your database? And what's the accuracy of your DFT data?

**Comment by Mskater** [comment]
Thank ***@***.***
How does one calculate what sel should be?
My database is simply the Silicon folder given on the deepmd website, with
9 sets, the last one for testing. It has 432 Si atoms. I counted them in
the type.raw file.
I don’t have the DFT data.  I’m just beginning to learn all this, as a grad
student. I appreciate your help very much.

**Comment by tuoping** [comment]
"sel" means "selected number of neighbors of center i". It doesn't mean total number of atoms in the system. "sel" is related to rcut.
Please refer to https://docs.deepmodeling.org/projects/deepmd/en/latest/troubleshooting/howtoset_sel.html

**Comment by Mskater** [comment]
Thank you @tuoping
I read that.  I had calculated that for the densest silicon, there would be
45 neighbor atoms in a radius of 6A rcut.  You say 46 is too small.
How did you arrive at 300 recommended for sel in this system?
 I don't see any explicit way to calculate from the link you point to.
Thanks so much.  🙏

**Comment by tuoping** [comment]
I apologize. Your method is definitely the best. Calculate the number of neighbors first and set "sel" according to the results. I can't see any other problem in your input file. But the error of force does look too large. I'm calling for help. @AnguseZhang

**Comment by njzjz** [comment]
Can you use logarithmic axis for both x- and y-axis?

**Comment by AnguseZhang** [comment]
I don't think there is problem with training parameters.
You said that you obtained data from DeePMD's website, then you can first compare your training accuracy with original paper. On the other hand, the model quality is system-specific. Then, to check the quality of your model, you should do some survey, such as finding some related papers and compare with them. Furthermore, you should test some physical properties, more than RMSE, like equation of states, elastic modulus, etc. Last but not least, I don't think the data set you used is enough for production. You can refer to DP-GEN and see how an accurate Deep Potential can be generated.

Yuzhi

**Reply by Mskater** [reply]
Thank you very much @AnguseZhang for getting involved in this question.
So, do you mean that the large error in forces can be due to the Si data not being
comprehensive enough for production?
And the surveying, are they the calculations to be done in LAMMPS?
Also I'll have to educate myself about DP-GEN.  I don't know how it relates to DeePMD-kit?

**Comment by tuoping** [comment]
May I ask which website is it? I failed to find it.

**Reply by njzjz** [reply]
I think it's http://www.deepmd.org/database/deeppot-se-data/. But I'm not sure if it is still maintained.

**Comment by vsa1920** [comment]
Hi! I am working on NNPs as a project. Would anyone be able to point me to a source where I can get data for Si as well as a trained NNP model? It will be really helpful. Thanks!

**Reply by njzjz** [reply]
Please fetch from https://aissquare.com/ or https://github.com/deepmodeling/AIS-Square.

**Reply by vsa1920** [reply]
> Please fetch from https://aissquare.com/ or https://github.com/deepmodeling/AIS-Square.

Thank you so much for your response! I extensively went through AIS Square but could not get a trained model for Silicon. I did get single solid phase data for Crystalline Silicon as given [here](https://www.aissquare.com/datasets/detail?pageType=datasets&name=deepmd-se-dataset).
